src/models/PackageVersion.js

```diff
@@ -60,6 +60,26 @@ class PackageVersion extends BaseModel {
 		return sql.select([ `${FileHits.table}.*`, `${File.table}.filename` ]);
 	}
 
+	static async getMostUsedFile (type, name, version) {
+		return db(this.table)
+			.select([
+				`${Package.table}.id`,
+				`${Package.table}.type`,
+				`${Package.table}.name`,
+				`${this.table}.version`,
+				`${File.table}.filename`,
+				`${FileHits.table}.hits`,
+			])
+			.where(`${Package.table}.type`, type)
+			.andWhere(`${Package.table}.name`, name)
+			.andWhere(`${this.table}.version`, version)
+			.join(Package.table, `${this.table}.packageId`, '=', `${Package.table}.id`)
+			.join(File.table, `${this.table}.id`, '=', `${File.table}.packageVersionId`)
+			.join(FileHits.table, `${File.table}.id`, '=', `${FileHits.table}.fileId`)
+			.orderBy(`${FileHits.table}.hits`, 'desc')
+			.first();
```

Comment:
This will need group by + sum to consider all hits, not just one day. It'll also need to consider file extensions.

I'm also thinking it might be worth making a materialized view for this to avoid computing it at runtime (see e.g., `view_top_packages_month` + `updateViewTopPackagesMonthupdateViewTopPackagesMonth`).

Reply:
This might need a regex in the end because I also wanted to exclude some files here. For a start, anything in top-level directories:
 - doc, docs, documentation
 - example, examples, sample, sample
 - test, tests,
 - cjs, esm, es6, es

Also, all files starting with a dot.

Reply:
Alternatively, we could consider setting a minimum number of hits - e.g., if the most popular file is JS and gets 1k hits, the CSS file must have at least 10% of that to be considered. The goal is to reduce cases when a package only has either CSS or JS but not both and we pick a random file of the other type from a folder like `documentation` because it also got a few hits per month.

Reply:
>I also wanted to exclude some files here

Makes sense. Maybe we should try to filter this while building the materialized view for most-used files in a package?

>Alternatively, we could consider setting a minimum number of hits

It's not really an alternative to the previous statement as I don't think this will work for not-so-popular packages where the main js could have 15 hits and it will be enough to have only 2 hits for the alternative file to be considered. So excluding irrelevant folders and files should work better. With those excluded, it probably makes not much sense to filter by hits count.

Reply:
> Makes sense. Maybe we should try to filter this while building the materialized view for most-used files in a package?

Yes, everything should be done while building the view.

> So excluding irrelevant folders and files should work better. With those excluded, it probably makes not much sense to filter by hits count.

👍